src/MsgProcessing.chpl

```diff
@@ -339,11 +340,38 @@ module MsgProcessing
                 repMsg = "set %s to %t".format(name, val);
             }
             when (DType.UInt64, DType.UInt64) {
+                // TODO create a .getBigIntValue()
+                // for now we treat this as a uint
                 var e = toSymEntry(gEnt,uint);
                 var val: uint = value.getUIntValue();
                 e.a = val;
                 repMsg = "set %s to %t".format(name, val);
             }
+            when (DType.BigInt, DType.BigInt) {
+                var e = toSymEntry(gEnt,bigint);
+                var val: uint = value.getUIntValue();
+                e.a = val:bigint;
+                repMsg = "set %s to %t".format(name, val);
+            }
+            when (DType.BigInt, DType.UInt64) {
+                var e = toSymEntry(gEnt,bigint);
+                var val: uint = value.getUIntValue();
+                e.a = val:bigint;
+                repMsg = "set %s to %t".format(name, val);
+            }
+            when (DType.BigInt, DType.Int64) {
+                var e = toSymEntry(gEnt,bigint);
+                var val: int = value.getIntValue();
+                e.a = val:bigint;
+                repMsg = "set %s to %t".format(name, val);
+            }
+            when (DType.BigInt, DType.Bool) {
+                var e = toSymEntry(gEnt,bigint);
+                var val: bool = value.getBoolValue();
+                // can't cast from a bool to a bigint, so first cast to int
+                e.a = val:int:bigint;
```

Comment:
I would recommend marking this for update in the future like you do at other locations so we do not miss it here.